brigade/views.py

```diff
@@ -296,8 +303,91 @@ def project_monitor(brigadeid=None):
     return render_template('monitor.html', projects=projects_with_tests, org_name=brigadeid)
 
 
-@redirect_from('/brigade/', '/brigade/list')
+@redirect_from('/brigade/list', '/brigades/')
+@app.route('/brigades')
+def brigade_list():
+    state_names = {
+        'AK': 'Alaska',
+        'AL': 'Alabama',
+        'AR': 'Arkansas',
+        'AS': 'American Samoa',
+        'AZ': 'Arizona',
+        'CA': 'California',
+        'CO': 'Colorado',
+        'CT': 'Connecticut',
+        'DC': 'District of Columbia',
+        'DE': 'Delaware',
+        'FL': 'Florida',
+        'GA': 'Georgia',
+        'GU': 'Guam',
+        'HI': 'Hawaii',
+        'IA': 'Iowa',
+        'ID': 'Idaho',
+        'IL': 'Illinois',
+        'IN': 'Indiana',
+        'KS': 'Kansas',
+        'KY': 'Kentucky',
+        'LA': 'Louisiana',
+        'MA': 'Massachusetts',
+        'MD': 'Maryland',
+        'ME': 'Maine',
+        'MI': 'Michigan',
+        'MN': 'Minnesota',
+        'MO': 'Missouri',
+        'MP': 'Northern Mariana Islands',
+        'MS': 'Mississippi',
+        'MT': 'Montana',
+        'NA': 'National',
+        'NC': 'North Carolina',
+        'ND': 'North Dakota',
+        'NE': 'Nebraska',
+        'NH': 'New Hampshire',
+        'NJ': 'New Jersey',
+        'NM': 'New Mexico',
+        'NV': 'Nevada',
+        'NY': 'New York',
+        'OH': 'Ohio',
+        'OK': 'Oklahoma',
+        'OR': 'Oregon',
+        'PA': 'Pennsylvania',
+        'PR': 'Puerto Rico',
+        'RI': 'Rhode Island',
+        'SC': 'South Carolina',
+        'SD': 'South Dakota',
+        'TN': 'Tennessee',
+        'TX': 'Texas',
+        'UT': 'Utah',
+        'VA': 'Virginia',
+        'VI': 'Virgin Islands',
+        'VT': 'Vermont',
+        'WA': 'Washington',
+        'WI': 'Wisconsin',
+        'WV': 'West Virginia',
+        'WY': 'Wyoming'
+    }
+    states = {}
+    brigades = cfapi.get_brigades(official_brigades_only=True)
+    brigades_total = len(brigades)
+    # Find all two-letter state abbreviations in the brigade's city, and add brigade to those states
+    for brigade in brigades:
+        brigade_name = brigade['properties']['name']
+        brigade_properties = {
+            'id': brigade['properties']['id'],
+            'city': brigade['properties']['city']
+        }
+        brigade_states = re.findall(
+            r'\b([A-Z]{2})\b', brigade['properties']['city'])
+        for state in brigade_states:
+            state_fullname = state_names[state]
+            if state_fullname not in states:
+                states[state_fullname] = {}
+            states[state_fullname][brigade_name] = brigade_properties
```

Comment:
General software engineering nitpick – doing logic like this in the view method makes it very difficult to test. If you instead add a method to `cfapi` such as `cfapi.get_official_brigades_by_state` or something, then it would be easier to test, and likely you could just copy-paste some other test and change very little.

Ideally, the view method just collects data and passes it through to the view with as little logic as possible.